### DISCUSSION TITLE
Looking for tips on using CN to restore low quality images without inducing bias from the SD model (answered)

### DISCUSSION THREAD

**Discussion by madrooky** [lllyasviel/ControlNet#565] [Q&A]
//edit: 
I wrote such a long text and everything, but there is one simple solution. There is a hidden feature for img2img, "inpainting conditioning mask strength", found out while playing with the photoshop plugin. When set to zero the image structure remains untouched. You need an inpainting model for that to work. I have no idea why, but the slider is hidden in settings/img2img right at the top. Default is 1, this means the image is free to change its form, if you pull it down to 0, the structure of the image remains largely untouched. This is perfect for a complete retouch or restauration. It can't really safe very fine nuances like wrinkles on the skin, but with Photoshop you can use the original picture with a layer mask over the new one, use the mode luminosity, create a negative layer mask and paint over the fine details you want to keep. That goes for fine skin structue and wrinkles in a face and does not change the new texture. It is simply awesome and I have no idea why this thing is hidden in the options for a year by now....

Here super easy example i put almost no effort in.

Input:
![image](https://github.com/lllyasviel/ControlNet/assets/131664307/3b845bce-7c7f-4c9d-bd78-370f51413d3f)

img2img, denoise 0.5, ICMS 0

![image](https://github.com/lllyasviel/ControlNet/assets/131664307/9e50bff6-00d4-457a-9803-eebedb45b708)

negative layer mask in luminocity mode to the the fine wrinkles back on the forehead
![image](https://github.com/lllyasviel/ControlNet/assets/131664307/46f0de37-71f1-4c21-b225-2ff11e620985)

**Comment by BNP1111** [comment]
Hi, are you processing in img2img mode or inpainting mode, thanks!

**Reply by madrooky** [reply]
The "inpainting conditioning mask strength" works in both img2img and inpaint, but the example above was made in img2img.
